privacyidea/static/components/token/controllers/tokenDetailController.js

```diff
@@ -190,9 +190,42 @@ myApp.controller("tokenDetailController", function ($scope,
         }, $scope.get);
     };
 
-    $scope.delete = function () {
-        TokenFactory.delete($scope.tokenSerial, $scope.return_to);
+    $scope.deleteTokenAsk = function() {
+            $('#dialogTokenDelete').modal();
+        var tokenType = $scope.token.info.tokenkind;
+        if (tokenType == "hardware"){
+        $('#dialogTokenDelete').modal();
+             } else {
+             TokenFactory.delete($scope.tokenSerial, $scope.return_to,
+                function (data) {
+                    if (data.result.value==true) {
+                        inform.add(gettextCatalog.getString("Hardware Token deleted " +
+                            "successfully."),
+                                    {type: "info"});
+                        $scope._getUsers();
+                        $location.path("/user/list");
+                     } else {
+                inform.add(gettextCatalog.getString("Failed to delete Token."), {type: "danger"});
+            };
+              });
     };
+    };
+
+    $scope.deleteToken = function(){
```

Comment:
This function is pointless. `$scope.delete()` already does the exact same thing. Please use $scope.delete().